server/src/com/sixsq/slipstream/ssclj/resources/session_jwt.clj

```diff
@@ -0,0 +1,89 @@
+(ns com.sixsq.slipstream.ssclj.resources.session-jwt
+  (:require
+    [clojure.tools.logging :as log]
+    [com.sixsq.slipstream.auth.cookies :as cookies]
+    [com.sixsq.slipstream.auth.internal :as auth-internal]
+    [com.sixsq.slipstream.auth.utils.timestamp :as ts]
+    [com.sixsq.slipstream.ssclj.resources.common.std-crud :as std-crud]
+    [com.sixsq.slipstream.ssclj.resources.common.utils :as u]
+    [com.sixsq.slipstream.ssclj.resources.session :as p]
+    [com.sixsq.slipstream.ssclj.resources.session-jwt.utils :as jwt-utils]
+    [com.sixsq.slipstream.ssclj.resources.session-template-api-key :as tpl]
+    [com.sixsq.slipstream.ssclj.resources.session.utils :as sutils]
+    [com.sixsq.slipstream.ssclj.resources.spec.session :as session]
+    [com.sixsq.slipstream.ssclj.resources.spec.session-template-jwt :as session-tpl]))
+
+
+(def ^:const authn-method "jwt")
+
+
+;;
+;; schemas
+;;
+
+(def SessionDescription
+  tpl/desc)
+
+
+;;
+;; description
+;;
+(def ^:const desc SessionDescription)
+
+
+;;
+;; multimethods for validation
+;;
+
+(def validate-fn (u/create-spec-validation-fn ::session/session))
+(defmethod p/validate-subtype authn-method
+  [resource]
+  (validate-fn resource))
+
+
+(def create-validate-fn (u/create-spec-validation-fn ::session-tpl/schema-create))
+(defmethod p/create-validate-subtype authn-method
+  [resource]
+  (create-validate-fn resource))
+
+
+;;
+;; transform template into session resource
+;;
+
+(defmethod p/tpl->session authn-method
+  [{:keys [token href] :as resource} {:keys [headers] :as request}]
+  (if token
+    (if-let [{:keys [iss]} (jwt-utils/extract-claims token)]
+      (if iss
+        (let [validation-response (jwt-utils/validate-jwt token)]
+          (if (= "OK" validation-response)
+            (if-let [matched-user (jwt-utils/match-user iss)]
```

Comment:
does this mean CIMI accepts logins for any user passed within the token without double checking with the internal users?

Reply:
Yes.  This was the reason for the comment: "Currently, no matching user is required, so any valid JWT will allow for a valid session to be created. If user matching is required, then I'll need more information on the requirements for the user resources associated with a device-id."